### PR TITLE
EP-4897: dev env for multiple WP/PHP versions and local build scripts

### PR DESCRIPTION
**Pull request by sergunich**
- Poetry for local scripts and dependencies
- Adds dev env for multiple WP/PHP versions for development and testing
- Fixures for initial WP data
- Autobuilds for Affiliate blocks
- Add git-hooks for lining that works out of the box
- Updated docs on how to "live"-develop plugin with SWP

### CODE REVIEW

**Comment by sergunich** [comment]
I'll merge it right away to unblock the following work but let me know if you have any questions or concerns

File: README.md [hunk 1]
```diff
@@ -1,7 +1,89 @@
...
+    volumes:
+      - ./wordpress-plugin/src:/var/www/html/web/app/mu-plugins/wordpress-plugin:ro
+```
+Or for tesing with result of `./dev.py build-zip`
+```yaml
+services:
+  solutions-wordpress:
+    volumes:
+      - ./wordpress-plugin/build/organic:/var/www/html/web/app/mu-plugins/wordpress-plugin:ro
+```
```
Comment (Mwindo) [inline README.md:83]: 🙌

File: README.md [hunk 2]
```diff
@@ -1,7 +1,89 @@
...
+$ poetry run ./dev.py down --help
+```
+
+## Organic Affiliate Features Development
+The Wordpress Plugin includes the following Affiliate App features:
+* Insert Product Card: implemented as a [Gutenberg Block](https://developer.wordpress.org/block-editor/getting-started/create-block/).
+* Insert Product Carousel: implemented as a [Gutenberg Block](https://developer.wordpress.org/block-editor/getting-started/create-block/).
+* Insert Affiliate Link: implemented as a [Custom Format](https://developer.wordpress.org/block-editor/how-to-guides/format-api/).
+
+These blocks are atomatically build on every file change after `./dev.py up` - see `docker compose logs -f --tail 20 nodejs` for logs
```
Comment (Mwindo) [inline README.md:58]: Super minor, but when we have time, just two small typos: "These blocks are automatically built on every file change after"

File: .env.template [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# By default we are using `Organic Demo` site from `Organic Demo` organization
+ORGANIC_DEMO_SITE_UUID='1579aae8-85e4-4ca7-b556-7b52816f529c'
+
+# TODO: FILL IT and uncomment!
+# Get API key from https://app.organic.ly/apps/organic-ads/documentation (switch to "Organic Demo" org/site)
+# ORGANIC_DEMO_SITE_APIKEY=''
+
+# TODO: FILL IT and uncomment!
+# https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic
+# COMPOSER_AUTH='{"github-oauth":{"github.com":"your_personal_github_token_here"}}'
```
Comment (Mwindo) [inline .env.template:22]: Probably want something in the README about this?

**Review by Mwindo — COMMENTED** [review]
Had to fiddle with Poetry a bit (https://stackoverflow.com/questions/42098126/mac-osx-python-ssl-sslerror-ssl-certificate-verify-failed-certificate-verify, https://stackoverflow.com/questions/73876790/poetry-configuration-is-invalid-additional-properties-are-not-allowed-group) and update to python 3.11, but things seemed to work as expected afterwards :)

Excellent work @sergunich !

File: .env.template (hunk 3 above)
Comment (Mwindo) [reply .env.template:22]: (In addition to the nice cli notice given when running the `up` command)

**Pull request merged**